app/controllers/api/v1/application_controller.rb

```
@@ -35,23 +44,36 @@ def user_not_authorized(exception)
+            subdomain = $1
+            Rails.logger.debug "Fallback subdomain from host: #{subdomain}"
+          end
+        end
+
         @organization = Organization.find_by(subdomain: subdomain)
         unless @organization
-          render json: { error: "Organization not found for subdomain: #{subdomain}" }, status: :not_found
+          Rails.logger.debug "Organization not found for subdomain: #{subdomain || 'none'}"
+          render json: { error: "Organization not found for subdomain: #{subdomain || 'none'}" }, status: :not_found
         end
       end
```

RuboCop (1.69.1)

[convention] Assignment Branch Condition size for set_organization_from_subdomain is too high. [<4, 22, 7> 23.43/23]

(Metrics/AbcSize)